VikingQuiz.Api/Controllers/QuizzesController.cs

```diff
@@ -70,36 +75,41 @@ public class QuizzesController : Controller
         [HttpPost]
         [RequestSizeLimit(5_000_000)]
         [Authorize]
-        public async Task<IActionResult> AddQuiz(NewQuizViewModel quizBodyData) {
+        public async Task<IActionResult> AddQuiz(NewQuizViewModel quizBodyData)
+        {
             int currentUserId = User.Claims.GetUserId();
             IActionResult imageHttpResponse = FileValidityChecker(quizBodyData);
-            if (IActionResult.Equals(imageHttpResponse, Ok())) {
+            if (IActionResult.Equals(imageHttpResponse, Ok()))
+            {
                 return imageHttpResponse;
             }
 
             string fileUrl;
             AzureBlobService blobService;
-            try {
-                blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
+            try
+            {
+                blobService = new AzureBlobService(azureContainerName);
+                fileUrl = await blobService.UploadPhotoAsync(quizBodyData.Files[0]);
             }
-            catch (Exception) {
+            catch (Exception)
+            {
                 return BadRequest("Image could not be uploaded.");
             }
 
 
-            Quiz createdQuiz = quizRepository.CreateQuiz(new Quiz {
+            Quiz createdQuiz = quizRepository.CreateQuiz(new Quiz
+            {
                 Title = quizBodyData.Title,
                 PictureUrl = fileUrl,
                 UserId = currentUserId
             });
-            if (createdQuiz == null) {
+            if (createdQuiz == null)
+            {
                 return BadRequest("Quiz couldn't be created");
             }
 
             QuizViewModel quizVm = entityToVmMapper.Map(createdQuiz);
-            quizVm.PictureUrl = blobService.urlPath.AbsoluteUri.ToString() + "users/" + fileUrl;
+            quizVm.PictureUrl = new AzureBlobService(azureContainerName).GetFullUrlOfFileName(quizVm.PictureUrl);
 
             return Ok(new { quizVm.Id, quizVm.Title, quizVm.PictureUrl });
```

Comment:
I don't understand why you are returning a new quizVm. Maybe you could just return quizVm, like return Ok(quizVm).